### ISSUE TITLE
badge number not increasing nor decreasing after put on test flight

### ISSUE THREAD

**Issue by abadikaka** [smooch/smooch-ios#145]
hi guys.. yesterday i sccesfully get the notif background foreground and the badge number working perfect. then i move it to testflight, use other certificate put new one also in smooch but suddenly the app badge number not increasing nor decreasing, only can receive notif background and foreground . What's wrong with it ? is it cert issues or code ? 

thanks

- smooch version 5.5.2

**Comment by wmora** [comment]
Hi @abadikaka can you confirm that when you uploaded the new certificate you enabled the option to `Use unread count as application badge number`? If so, opening the conversation view should reset the badge number.

<img width="826" alt="screen shot 2017-04-04 at 9 02 37 am" src="https://cloud.githubusercontent.com/assets/2448046/24657790/a5eedcb8-1915-11e7-9ca8-612668102f74.png">

**Comment by abadikaka** [comment]
Hi @wmora yes it found that my supervisor hasnt ticked that one.. thanks
But i have another issue regarding unread msg number.

I am trying get the list of conv and the unread message, then after that i open the chat, but i didnt reply the chat only see the chat.. After that i reload it again, the unread count become 0 ( correct behavior ). But once i logout the smooch with that userid, then login again with that userId, and ask for conversationlist again, i got the unread count back to previous unread count...

the unread count will be correct if i reply it when i got the chat previously. Is that a known issue ?

**Comment by abadikaka** [comment]
Hi @wmora yes it found that my supervisor hasnt ticked that one.. thanks
But i have another issue regarding unread msg number.

I am trying get the list of conv and the unread message, then after that i open the chat, but i didnt reply the chat only see the chat.. After that i reload it again, the unread count become 0 ( correct behavior ). But once i logout the smooch with that userid, then login again with that userId, and ask for conversationlist again, i got the unread count back to previous unread count...

the unread count will be correct if i reply it when i got the chat previously. Is that a known issue ?

**Comment by wmora** [comment]
hmmm I'm not able to reproduce and we don't have a known issue for it. Can you reproduce consistently so you can list the use case step by step?

**Comment by abadikaka** [comment]
@wmora  ok. i will also give u the json i created using smooch response.

1. Login using some userID, and sent a message from businness integration, we get the notifications , open it but dont reply it..
```

{
      "icon" : "https:\/\/app.smooch.io\/v1\/appmakers\/57f8bf8832a9173f00b1f4b1\/avatar",
      "channel_name" : "Shrouk Khan",
      "last_message" : "Hey , can I ask some wine",
      "unread_messages" : 2,
      "app_token" : "3afzpr4hogs8z5znmomda0ko5",
      "time_since_last_message" : "30 seconds"
    }
```

2. I have a button to refresh getConversationList again, and got correct response with unread messages 0
```

{
      "icon" : "https:\/\/app.smooch.io\/v1\/appmakers\/57f8bf8832a9173f00b1f4b1\/avatar",
      "channel_name" : "Shrouk Khan",
      "last_message" : "Hey , can I ask some wine",
      "unread_messages" : 0,
      "app_token" : "3afzpr4hogs8z5znmomda0ko5",
      "time_since_last_message" : "55 seconds"
    }
```

3. I do logout that userID, and then log in again with that userID and do step 1 again or ask for getConversationList again , i get the unread messages back to 2. it should be 0
```

 {
      "icon" : "https:\/\/app.smooch.io\/v1\/appmakers\/57f8bf8832a9173f00b1f4b1\/avatar",
      "channel_name" : "Shrouk Khan",
      "last_message" : "Hey , can I ask some wine",
      "unread_messages" : 2,
      "app_token" : "3afzpr4hogs8z5znmomda0ko5",
      "time_since_last_message" : "3 minutes"
    }
```

But if i on step 1 , i reply back to business integration, it goes correctly to 0 (unread messages)

That is what i found, i dont know it is an issue or maybe that is the rule that i need to accept to get unread msg

**Comment by abadikaka** [comment]
anyway if u direclty log out and log in again,u will see the unread not 0 again.. in this case if u dont reply the chat when u open it.. @wmora  
thanks for ur help.. Sorry my reply kinda late, different time zones :D

**Comment by wmora** [comment]
thanks for the detail @abadikaka it seems to be the same issue reported in #148

**Comment by abadikaka** [comment]
@wmora okay thank you so much . inform to me as well if the issue has been fixed in this thread.. hope my explanation can help you..

**Comment by abadikaka** [comment]
Hi @wmora is there a way to retrieve the app name of the apptoken ?

**Comment by wmora** [comment]
@abadikaka not from the SDK, but you can use our [rest API](https://docs.smooch.io/rest/#get-app) for that

**Comment by wmora** [comment]
@abadikaka the unread count has been fixed in [v5.6.2](https://github.com/smooch/smooch-ios/releases/tag/5.6.2)

**Comment by abadikaka** [comment]
@wmora or @mspensieri for localization i want to ask, is it possible to change the title of conversation using parameter on the localization ?
i mean i success using SmoochLocalizable.string. but i want to make it dynamic base on the channel name..
so i put

"Message" = "%@"

can i do this for the navigation title item on nav bar itself for using dynamic parameter on the localization ? Thanks
